rct229/rulesets/ashrae9012019/section19/section19rule21.py

```diff
@@ -0,0 +1,510 @@
+                        if lighting_space_type_p == LIGHTING_SPACE.FOOD_PREPARATION_AREA
+                        else False
+                    )
+                else:
+                    all_lighting_space_types_defined = False
+
+                if space_p.get("ventilation_space_type") is not None:
+                    ventilation_space_type_p = space_p["ventilation_space_type"]
+                    serves_zones_with_systems_likely_exhausting_toxic_etc = (
+                        True
+                        if ventilation_space_type_p
+                        == LIGHTING_SPACE.MISCELLANEOUS_SPACES_MANUFACTURING_WHERE_HAZARDOUS_MATERIALS_ARE_USED_EXCLUDES_HEAVY_INDUSTRIAL_AND_CHEMICAL_PROCESSES
+                        else False
+                    )
+
+                    serves_kitchen_space = (
+                        True
+                        if ventilation_space_type_p
+                        == LIGHTING_SPACE.FOOD_AND_BEVERAGE_SERVICE_KITCHEN_COOKING
+                        else False
+                    )
+                else:
+                    all_ventilation_space_types_defined = False
+
+            hvac_sys_and_assoc_zones_largest_exhaust_source[
+                hvac_id_b
+            ] = get_hvac_sys_and_assoc_zones_largest_exhaust_source(rmi_b, hvac_id_b)
+
+        for hvac_id_p in get_list_hvac_systems_associated_with_zone(rmi_p, zone_id_b):
+            hvac_p = find_one(
+                f"$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[?(@.id == {hvac_id_p})]",
+                rmi_p,
+            )
+            ER_modeled_in_proposed = (
+                True
+                if (
+                    getattr_(hvac_p, "HVAC", "fan_system").get("air_energy_recovery")
+                    is not None
+                    and getattr_(
+                        hvac_p,
+                        "HVAC",
+                        "fan_system",
+                        "air_energy_recovery",
+                        "energy_recovery_type",
+                    )
+                    != ENERGY_RECOVERY.NONE
+                )
+                else False
+            )
+
+            serves_zones_that_have_dehumid_heat_recovery = (
+                True
+                if (
+                    hvac_p.get("cooling_system") is not None
+                    and hvac_p["cooling_system"].get("dehumidification_type")
+                    == DEHUMIDIFICATION.SERIES_HEAT_RECOVERY
+                )
+                else False
+            )
+
+        sys_type_heating_only = (
+            True
+            if any(
+                [
+                    baseline_system_type_compare(
+                        system_type, applicable_sys_type, False
+                    )
+                    for system_type in get_baseline_system_types(rmi_b).keys()
+                    for applicable_sys_type in APPLICABLE_SYS_TYPES
+                ]
+            )
+            else False
+        )
```

Comment:
I thought this should be specific to the system, not the all the systems?

```python
baseline_system_types = get_baseline_system_types(rmi_b)
sys_type_heating_only = any([hvac_id_p in baseline_system_teyps for applicable_sys_type in APPLICABLE_SYS_TYPES])
```

Reply:
Addressed.